code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -1562,46 +1562,95 @@ All effects don't start immediately, but rather get worse over time; the rate is
+			mighty_shield = theshield
+			mighty_shield.block_chance += 10
+			to_chat(thehuman, "<span class='notice'>[theshield] appears polished, although you don't recall polishing it.</span>")
+			return TRUE
+
+/datum/reagent/consumable/ethanol/alexander/on_mob_life(mob/living/L)
+	if(mighty_shield && !(mighty_shield in L.contents)) //If you had a shield and lose it, you lose the reagent as well. Otherwise this is just a normal drink.
+		L.reagents.del_reagent("alexander")
+	else
+		..()
+
+/datum/reagent/consumable/ethanol/alexander/on_mob_delete(mob/living/L)
+	if(mighty_shield)
+		mighty_shield.block_chance -= 10
+		to_chat(L,"<span class='notice'>You notice [mighty_shield] looks worn again. Weird.</span>")
+	..()
+
 
 /datum/reagent/consumable/ethanol/sidecar
 	name = "Sidecar"
 	id = "sidecar"
-	description = "The one ride you’ll gladly give up the wheel for."
+	description = "The one ride you'll gladly give up the wheel for."
 	color = "#FFC55B"
 	boozepwr = 80
 	taste_description = "delicious freedom"
 	glass_icon_state = "sidecar"
 	glass_name = "Sidecar"
-	glass_desc = "The one ride you’ll gladly give up the wheel for."
+	glass_desc = "The one ride you'll gladly give up the wheel for."
 
 /datum/reagent/consumable/ethanol/between_the_sheets
 	name = "Between the Sheets"
 	id = "between_the_sheets"
-	description = "A provocatively named classic."
+	description = "A provocatively named classic. Funny enough, doctors recommend drinking it before taking a nap."
 	color = "#F4C35A"
 	boozepwr = 80
 	taste_description = "seduction"
 	glass_icon_state = "between_the_sheets"
 	glass_name = "Between the Sheets"
-	glass_desc = "A provocatively named classic."
+	glass_desc = "The only drink that comes with a label reminding you of Nanotrasen's zero-tolerance promiscuity policy."
+
+/datum/reagent/consumable/ethanol/between_the_sheets/on_mob_life(mob/living/L)
+	if(L.IsSleeping())
+		if(L.bruteloss && L.fireloss) //If you are damaged by both types, slightly increased healing but it only heals one. The more the merrier wink wink.
+			if(prob(50))
+				L.adjustBruteLoss(-0.25)
+			else
+				L.adjustFireLoss(-0.25)
+		else if(L.bruteloss && !L.fireloss) //If you have only one, it still heals but not as well.
+			L.adjustBruteLoss(-0.2)
+		else if(!L.bruteloss && L.fireloss)
+			L.adjustFireLoss(-0.2)
 
 /datum/reagent/consumable/ethanol/kamikaze
 	name = "Kamikaze"
 	id = "kamikaze"
-	description = "Divinely windy."
+	description = "Science has yet to understand the seemingly mystical properties of this drink. Not only does it damage the individual, but it also damages those nearby the individual can see."
 	color = "#EEF191"
 	boozepwr = 60
 	taste_description = "divine windiness"
 	glass_icon_state = "kamikaze"
 	glass_name = "Kamikaze"
-	glass_desc = "Divinely windy."
+	glass_desc = "A drink you don't exactly make friends with."
+	var/mutable_appearance/explosion_overlay
+
+/datum/reagent/consumable/ethanol/kamikaze/on_mob_add(mob/living/L)
```

Comment:
I believe there is a method to make this only work if you ingest it.

Reply:
errr this is only applicable on `reaction_mob` unfortunately,

I plan on making it like hearty punch where it takes a bunch to make a little.

Reply:
uhhh you can macro on the drink dispenser